### PR TITLE
Feature/3241 add cli option to specify log file path

### PR DESCRIPTION
**Pull request by randy-seng**
OP#3241

### CODE REVIEW

File: convert.py [hunk 1]
```diff
@@ -145,7 +152,7 @@ def _extract_paths(args: argparse.Namespace) -> list[Path]:
             "No paths have been passed as command line args."
             "No paths have been defined in the user config."
         )
-    paths = [Path(str_path) for str_path in str_paths]
+    paths = [Path(str_path).expanduser() for str_path in str_paths]
```
Comment (briemla) [inline convert.py:155]: Do we also need `resolve` here as in `check_if_all_paths_exist` in `files.py`?

File: detect.py [hunk 2]
```diff
@@ -210,7 +217,7 @@ def _extract_paths(args: argparse.Namespace) -> list[Path]:
             "No paths have been passed as command line args."
             "No paths have been defined in the user config."
         )
-    paths = [Path(str_path) for str_path in str_paths]
+    paths = [Path(str_path).expanduser() for str_path in str_paths]
```
Comment (briemla) [inline detect.py:220]: See `convert.py` line 155.

File: detect.py [hunk 3]
```diff
@@ -227,19 +234,12 @@ def _configure_logger(args: argparse.Namespace) -> logging.Logger:
...
-            raise
-    else:
-        log_dir = Path(args.log_dir)
-
     log.add_console_handler(level=log_level_console)
-
-    log.add_file_handler(log_dir=log_dir, level=log_level_file)
-
+    log.add_file_handler(
+        Path(args.logfile).expanduser(),
```
Comment (briemla) [inline detect.py:239]: See `convert.py` line 155.

File: convert.py [hunk 4]
```diff
@@ -162,19 +169,12 @@ def _configure_logger(args: argparse.Namespace) -> logging.Logger:
...
-            raise
-    else:
-        log_dir = Path(args.log_dir)
-
     log.add_console_handler(level=log_level_console)
-
-    log.add_file_handler(log_dir=log_dir, level=log_level_file)
-
+    log.add_file_handler(
+        Path(args.logfile).expanduser(),
```
Comment (briemla) [inline convert.py:174]: See `convert.py` line 155.

File: reference_points_picker.py [hunk 5]
```diff
@@ -64,7 +64,7 @@ def _configure_logger(args: argparse.Namespace) -> logging.Logger:
     else:
         log_dir = Path(args.log_dir)
 
-    log.add_file_handler(log_dir=log_dir, level=log_level_file)
+    log.add_file_handler(log_file=log_dir, level=log_level_file)
```
Comment (briemla) [inline reference_points_picker.py:67]: Until now, a directory has been handed over to the logger. Should we add a file name or use the default filename?

File: track.py [hunk 6]
```diff
@@ -165,7 +172,7 @@ def _extract_paths(args: argparse.Namespace) -> list[Path]:
             "No paths have been passed as command line args."
             "No paths have been defined in the user config."
         )
-    paths = [Path(str_path) for str_path in str_paths]
+    paths = [Path(str_path).expanduser() for str_path in str_paths]
```
Comment (briemla) [inline track.py:175]: See `convert.py` line 155.

File: track.py [hunk 7]
```diff
@@ -183,19 +190,12 @@ def _configure_logger(args: argparse.Namespace) -> logging.Logger:
...
-            raise
-    else:
-        log_dir = Path(args.log_dir)
-
     log.add_console_handler(level=log_level_console)
-
-    log.add_file_handler(log_dir=log_dir, level=log_level_file)
-
+    log.add_file_handler(
+        Path(args.logfile).expanduser(),
```
Comment (briemla) [inline track.py:195]: See `convert.py` line 155.

File: transform.py [hunk 8]
```diff
@@ -126,7 +134,7 @@ def _extract_paths(args: argparse.Namespace) -> list[Path]:
             "No paths have been passed as command line args."
             "No paths have been defined in the user config."
         )
-    paths = [Path(str_path) for str_path in str_paths]
+    paths = [Path(str_path).expanduser() for str_path in str_paths]
```
Comment (briemla) [inline transform.py:137]: See `convert.py` line 155.

File: transform.py [hunk 9]
```diff
@@ -143,19 +151,12 @@ def _configure_logger(args: argparse.Namespace) -> logging.Logger:
...
-            raise
-    else:
-        log_dir = Path(args.log_dir)
-
     log.add_console_handler(level=log_level_console)
-
-    log.add_file_handler(log_dir=log_dir, level=log_level_file)
-
+    log.add_file_handler(
+        Path(args.logfile).expanduser(),
```
Comment (briemla) [inline transform.py:156]: See `convert.py` line 155.

**Review by briemla — CHANGES_REQUESTED** [review]

File: reference_points_picker.py (hunk 5 above)
Comment (randy-seng) [reply reference_points_picker.py]: Yes, let us implement the same behavior. I simply forgot to add the CLI options `--logfile` and `logfile_overwrite` for the reference points picker.

**Review by briemla — APPROVED** [review]

File: convert.py (hunk 1 above)
Comment (randy-seng) [reply convert.py:155]: We do not necessarily need to use `resolve` the paths manually here. `pathlib` will be able to resolve the paths without the call to `resolve`. I will remove it in `check_if_all_paths_exist`.

**Pull request merged**